inline.h

```diff
@@ -705,6 +705,45 @@ Perl_rpp_replace_at(pTHX_ SV **sp, SV *sv)
 }
 
 
+/*
+=for apidoc rpp_context
+
+Impose void, scalar or list context on the stack.
+First, pop C<extra> items off the stack, then when C<gimme> is:
+C<G_LIST>:   return as-is.
+C<G_VOID>:   pop everything back to C<mark>
+C<G_SCALAR>: move the top stack item (or C<&PL_sv_undef> if none) to
+C<mark+1> and free everything above it.
+
+=cut
+*/
+
+PERL_STATIC_INLINE void
+Perl_rpp_context(pTHX_ SV **mark, U8 gimme, SSize_t extra)
+{
+    PERL_ARGS_ASSERT_RPP_CONTEXT;
+
```

Comment:
Would it be worth asserting that `mark <= PL_stack_sp - extra` ?

Possibly also `extra >= 0`.